tests/dragonfly/__init__.py

```diff
@@ -72,9 +99,13 @@ def stop(self, kill=False):
     def _start(self):
         if self.params.existing_port:
             return
+
+        if self.dynamic_port:
+            self._port = None
+
         base_args = [f"--{v}" for v in self.params.args]
         all_args = self.format_args(self.args) + base_args
-        print(f"Starting instance on {self.port} with arguments {all_args} from {self.params.path}")
+        print(f"Starting instance with arguments {all_args} from {self.params.path}")
```

Comment:
not a big expert, but I think use logging is better over print. 
or at least we should agree on this.